src/Tools/ExternalAccess/LiveShare/Completion/RoslynCompletionProvider.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.LanguageServer.CustomProtocol;
+using Microsoft.CodeAnalysis.Text;
+using Newtonsoft.Json.Linq;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.LiveShare.Completion
+{
+    internal class RoslynCompletionProvider : CommonCompletionProvider
+    {
+        private readonly RoslynLSPClientServiceFactory _roslynLSPClientServiceFactory;
+
+        public RoslynCompletionProvider(RoslynLSPClientServiceFactory roslynLSPClientServiceFactory)
+        {
+            _roslynLSPClientServiceFactory = roslynLSPClientServiceFactory ?? throw new ArgumentNullException(nameof(roslynLSPClientServiceFactory));
+        }
+
+        public override async Task ProvideCompletionsAsync(CompletionContext context)
+        {
+            // This provider is exported for all workspaces - so limit it to just our workspace & the debugger's intellisense workspace
+            if (context.Document.Project.Solution.Workspace.Kind != WorkspaceKind.AnyCodeRoslynWorkspace &&
+                context.Document.Project.Solution.Workspace.Kind != StringConstants.DebuggerIntellisenseWorkspaceKind)
```

Comment:
seems very arbitrary. why some just checked AnyCode and some checks debugger as well. but, these will be removed so, probably okay.

Reply:
I'm not sure - I can dig into it but these will all hopefully be deleted soon.